### PR TITLE
conditional encode

### PR DESCRIPTION
**Pull request by roynalnaruto**
conditional encoding of batch -> blob

### CODE REVIEW

**Comment by lispc** [comment]
picked to https://github.com/scroll-tech/zkevm-circuits/pull/1396

**Review by lispc — APPROVED** [review]
Do we need another review?

File: aggregator/src/param.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ impl ConfigParams {
         Self {
             strategy: FpStrategy::Simple,
             degree: 21,
-            num_advice: vec![63],
+            num_advice: vec![100],
```
Comment (noel2004) [inline aggregator/src/param.rs]: I think we have just induced several cols but here the advice cols increase from 63 -> 100 ?

**Review by noel2004 — APPROVED** [review]
Everything looks good except for the `blob_circuit_completeness` test is blanked. What do we need to restore it?

**Comment by roynalnaruto** [comment]
@noel2004 

> except for the blob_circuit_completeness test is blanked

What do you mean?

The only test under `blob_circuit_completeness` that I have disabled currently is the "full blob" test, i.e. a batch that when encoded produces _exactly_ `4096 * 31` bytes. The reason to comment it out is that, after prepending the single byte (`enableEncoding`), the present test fails (as it produces +1 byte than "full").

We need to change `batch274` such that the assertion of "full" blob is still satisfied with this new structure of blob.

File: prover/src/types.rs [hunk 2]
```diff
@@ -45,6 +45,7 @@ impl ChunkProvingTask {
 pub struct BatchProvingTask {
     pub chunk_proofs: Vec<ChunkProof>,
     pub batch_header: BatchHeader<MAX_AGG_SNARKS>,
+    pub blob_bytes: Vec<u8>,
```
Comment (lispc) [inline prover/src/types.rs:49]: I may need add a base64 serde trait here.

**Comment by lispc** [comment]
where is the logic to `assert_eq!(decode(blob), compute_batch_bytes(chunk_infos))`?

**Comment by roynalnaruto** [comment]
@lispc 

> where is the logic to assert_eq!(decode(blob), compute_batch_bytes(chunk_infos))?

We might need to add more sanity checks

File: aggregator/src/param.rs (hunk 1 above)
Comment (roynalnaruto) [reply aggregator/src/param.rs]: I have reduced it to the minimum possible, that allows the integration tests to run.

**Review by lispc — APPROVED** [review]

**Pull request merged**